openscm/adapters/dice.py

```diff
@@ -0,0 +1,388 @@
+            )
+
+    def _initialize_model_input(self) -> None:
+        pass
+
+    def _initialize_run_parameters(self) -> None:
+        self._timestep = 0
+        self._timestep_count = (
+            self._stop_time - self._start_time
+        ) // self._period_length
+
+        time_points = create_time_points(
+            self._start_time,
+            self._period_length,
+            self._timestep_count,
+            ParameterType.POINT_TIMESERIES,
+        )
+        time_points_for_averages = create_time_points(
+            self._start_time,
+            self._period_length,
+            self._timestep_count,
+            ParameterType.AVERAGE_TIMESERIES,
+        )
+
+        # Total CO2 emissions (GtCO2 per year)
+        self._views.E = self._parameters.get_timeseries_view(
+            ("Emissions", "CO2"),
+            ("World",),
+            "GtCO2/a",
+            time_points_for_averages,
+            ParameterType.AVERAGE_TIMESERIES,
+            InterpolationType.LINEAR,
+            ExtrapolationType.LINEAR,
+        )
+
+        # TODO remove setting missing emissions once there is a method in tests for that
+        if self._views.E.is_empty:
+            self._parameters.get_writable_timeseries_view(
+                ("Emissions", "CO2"),
+                ("World",),
+                "GtCO2/a",
+                time_points_for_averages,
+                ParameterType.AVERAGE_TIMESERIES,
+                InterpolationType.LINEAR,
+                ExtrapolationType.LINEAR,
+            ).set(np.zeros(self._timestep_count))
+
+        # Concentration in atmosphere
+        self._views.M_atm = self._output.get_writable_timeseries_view(
+            ("Concentration", "Atmosphere"),
+            ("World",),
+            "GtC",
+            time_points,
+            ParameterType.POINT_TIMESERIES,
+            InterpolationType.LINEAR,
+            ExtrapolationType.LINEAR,
+        )
+
+        # Carbon concentration increase in lower oceans (rel. to 1750)
+        self._views.M_l = self._output.get_writable_timeseries_view(
+            ("Concentration", "Ocean", "lower"),
+            ("World",),
+            "GtC",
+            time_points,
+            ParameterType.POINT_TIMESERIES,
+            InterpolationType.LINEAR,
+            ExtrapolationType.LINEAR,
+        )
+
+        # Carbon concentration increase in shallow oceans (rel. to 1750)
+        self._views.M_u = self._output.get_writable_timeseries_view(
+            ("Concentration", "Ocean", "shallow"),
+            ("World",),
+            "GtC",
+            time_points,
+            ParameterType.POINT_TIMESERIES,
+            InterpolationType.LINEAR,
+            ExtrapolationType.LINEAR,
+        )
+
+        # Increase temperature of atmosphere (rel. to 1900)
+        self._views.T_atm = self._output.get_writable_timeseries_view(
+            ("Temperature Increase", "Atmosphere"),
```

Comment:
alter to follow #139

Reply:
i have a list of all i need to alter, no worries